x-pack/plugins/ui_actions_enhanced/server/types.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  PersistableState,
+  PersistableStateDefinition,
+  SerializableState,
+} from '../../../../src/plugins/kibana_utils/common';
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type SerializedAction<Config extends SerializableState = SerializableState> = {
```

Comment:
I know this is copied, but I'd really preferred it to be reused. I thinks in this case sharing the same type is crucial, because it represents stored entity. If this gets out of sync, then server and pubic implementation gets out of sync and we get into trouble.